test/ProgressOnderwijsUtils.Tests/Data/BulkInsertTest.cs

```diff
@@ -74,35 +73,56 @@ sealed class SampleRow : ValueBase<SampleRow>, IWrittenImplicitly, IReadImplicit
             }
         };
 
+        public static BulkInsertTestSampleRow[] SampleRows(int n)
+            => Enumerable.Range(0, (n + 3) / 4).SelectMany(i => FourSampleRows).ToArray();
+    }
+
+    public sealed class BulkInsertTest : TransactedLocalConnection
+    {
         [Fact]
         public void BulkCopysWithConcurrentQueriesCrash()
         {
-            var target = CreateTable(SQL($"#test"));
-            var evilEnumerable = SampleData.Where(o => SQL($"select 1").ReadScalar<int>(Connection) == 1);
+            var target = BulkInsertTestSampleRow.CreateTable(Connection, SQL($"#test"));
+            var evilEnumerable = BulkInsertTestSampleRow.SampleRows(16).Where(o => SQL($"select 1").ReadScalar<int>(Connection) == 1);
             Assert.ThrowsAny<Exception>(() => evilEnumerable.BulkCopyToSqlServer(Connection, target));
         }
 
         [Fact]
         public void BulkInsertAndReadRoundTrips()
         {
-            var target = CreateTable(SQL($"#test"));
-            SampleData.BulkCopyToSqlServer(Connection, target);
-            var fromDb = SQL($"select * from #test").ReadPocos<SampleRow>(Connection);
-            AssertCollectionsEquivalent(SampleData, fromDb);
+            var target = BulkInsertTestSampleRow.CreateTable(Connection, SQL($"#test"));
+            BulkInsertTestSampleRow.SampleRows(4).BulkCopyToSqlServer(Connection, target);
+            var fromDb = SQL($"select * from #test").ReadPocos<BulkInsertTestSampleRow>(Connection);
+            AssertCollectionsEquivalent(BulkInsertTestSampleRow.SampleRows(4), fromDb);
+        }
+
+        [Fact]
+        public void BulkInsertAndReadRoundTrips_ManyRows()

```

Comment:
Hebben we nog een test nodig voor de andere situatie wat betreft `enumerable is IReadOnlyList<T> || enumerable is ICollection<T>`?

Reply:
zijnde dat die gewoon de sequence onveranderd doorgeeft leek me dat minder boeiend.

Reply:
maar goed, een minimale test is niet moeilijk, toegevoegd.